pkg/credentialprovider/external/external.go

```diff
@@ -0,0 +1,175 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package external
+
+import (
+	"bytes"
+	"fmt"
+	"os/exec"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/pkg/credentialprovider"
+	"k8s.io/kubernetes/pkg/credentialprovider/apis/dockerauth"
+	"k8s.io/kubernetes/pkg/credentialprovider/apis/dockerauth/v1alpha1"
+)
+
+var (
+	scheme = runtime.NewScheme()
+	codecs = serializer.NewCodecFactory(scheme)
+)
+
+func init() {
+	dockerauth.AddToScheme(scheme)
+	v1alpha1.AddToScheme(scheme)
+}
+
+// Called from kubelet if alpha feature flag is enabled
+func RegisterExternalProviders(pluginConfigFile string) error {
+	externalProviders, err := NewExternalCredentialProvidersForConfig(pluginConfigFile)
+	if err != nil {
+		return err
+	}
+
+	for _, provider := range externalProviders {
+		credentialprovider.RegisterCredentialProvider(provider.name, provider)
+	}
+
+	return nil
+}
+
+func NewExternalCredentialProvidersForConfig(config string) ([]*externalCredentialProvider, error) {
```

Comment:
Do we want this to be public? Seems odd to have a public method with a private return type.